utils/post_processing/restructure_borealis.py

```diff
@@ -0,0 +1,507 @@
+	return data_dict
+
+def rawacf_site_to_array(data_record):
+	"""
+	Restructuring method for rawacf data
+	Args:
+		data_record		a timestamped record loaded from an
+		 				hdf5 Borealis rawacf data file
+	Returns:
+		 	data_dict:	A dictionary containing the data from data_record
+		 				reformatted to be stored entirely as arrays, or as
+		 				one entry if the field does not change between records
+	"""
+	data_dict = dict()
+	num_records = len(data_record)
+	max_seqs = find_max_sequences(data_record)
+
+	# write shared fields to dictionary
+	k = list(data_record.keys())[0]
+	for field in shared_rawacf:
+		data_dict[field] = data_record[k][field]
+
+	# handle unshared data fields
+	dims = data_dict["correlation_dimensions"]
+	num_beams = dims[0]
+	num_ranges = dims[1]
+	num_lags = dims[2]
+	data_shape = (num_records, num_beams, num_ranges, num_lags)
+
+	noise_buffer_offset = max_seqs
+	noise_buffer = np.zeros(num_records * max_seqs)
+	noise_shape = (num_records, max_seqs)
+
+	sqn_ts_array = np.empty((num_records, max_seqs))
+	sqn_num_array = np.empty(num_records)
+	main_array = np.empty(data_shape, dtype=np.complex64)
+	intf_array = np.empty(data_shape, dtype=np.complex64)
+	xcfs_array = np.empty(data_shape, dtype=np.complex64)
+
+	int_time_array = np.empty(num_records)
+	
+
+	rec_idx = 0
+	for k in data_record:
+		sqn_num_array[rec_idx] = data_record[k]["num_sequences"]
+		int_time_array[rec_idx] = data_record[k]["int_time"]
+
+		# some records have fewer than the specified number of sequences
+		# get around this by zero padding to the recorded number
+		sqn_timestamps = data_record[k]["sqn_timestamps"]
+		while len(sqn_timestamps) < max_seqs:
+			sqn_timestamps = np.append(sqn_timestamps, 0)
+		sqn_ts_array[rec_idx] = sqn_timestamps
+
+		rec_noise = data_record[k]["noise_at_freq"]
+		noise_pos = rec_idx * noise_buffer_offset
+		noise_end = noise_pos + data_record[k]["num_sequences"]
+		noise_buffer[noise_pos:noise_end] = rec_noise
+
+		data_dict["noise_at_freq"] = noise_buffer.reshape(noise_shape)
+
+		main_array[rec_idx] = data_record[k]["main_acfs"].reshape(dims)
+		intf_array[rec_idx] = data_record[k]["intf_acfs"].reshape(dims)
+		xcfs_array[rec_idx] = data_record[k]["xcfs"].reshape(dims)
+
+		rec_idx += 1
+
+	# write leftover metadata and data
+	data_dict["int_time"] = int_time_array
+	data_dict["sqn_timestamps"] = sqn_ts_array
+	data_dict["num_sequences"] = sqn_num_array
+	data_dict["noise_at_freq"] = noise_buffer.reshape(noise_shape)
+	data_dict["correlation_descriptors"] = np.insert(data_dict["correlation_descriptors"], 0, "num_records")
+
+	data_dict["main_acfs"] = main_array
+	data_dict["intf_acfs"] = intf_array
+	data_dict["xcfs"] = xcfs_array
+
+	return data_dict
+
+
+def write_array_format_data(data_dict, data_path):
+	"""
+	Writes a record of restructured borealis data to file.
+	Args:
+		data_dict:		A timestamped dictionary containing site
+						formatted Borealis data
+		data_path:		The path to the data file storing data_dict
```

Comment:
Since you're doing slicing (below with dd.io.save...) and assuming something about the format of the data_path, perhaps writing that here can be of help, or even doing a check that the filename is in the correct format and raising an error if that's not the case.